apps/web/src/app/api/v1/workspaces/[wsId]/products/[productId]/inventory/route.ts

```diff
@@ -0,0 +1,334 @@
+import { createClient } from '@tuturuuu/supabase/next/server';
+import type { ProductInventory } from '@tuturuuu/types/primitives/ProductInventory';
+import { NextResponse } from 'next/server';
+
+interface Params {
+  params: Promise<{
+    wsId: string;
+    productId: string;
+  }>;
+}
+
+interface RequestBody {
+  inventory: ProductInventory[];
+}
+
+export async function POST(req: Request, { params }: Params) {
+  const supabase = await createClient();
+  const { inventory } = (await req.json()) as RequestBody;
+  const { productId } = await params;
+
+  // Validate that product exists
+  const { data: product, error: productError } = await supabase
+    .from('workspace_products')
+    .select('id')
+    .eq('id', productId)
+    .single();
+
+  if (productError || !product) {
+    return NextResponse.json({ message: 'Product not found' }, { status: 404 });
+  }
+
+  // Insert inventory items
+  const { error } = await supabase.from('inventory_products').insert(
+    inventory.map((item) => ({
+      ...item,
+      product_id: productId,
+    }))
+  );
+
+  if (error) {
+    console.log(error);
+    return NextResponse.json(
+      { message: 'Error creating inventory' },
+      { status: 500 }
+    );
+  }
+
+  return NextResponse.json({ message: 'Inventory created successfully' });
+}
+
+export async function PATCH(req: Request, { params }: Params) {
+  const supabase = await createClient();
+  const {
+    data: { user },
+  } = await supabase.auth.getUser();
+  if (!user) {
+    return NextResponse.json({ message: 'Unauthorized' }, { status: 401 });
+  }
+  const { inventory } = (await req.json()) as RequestBody;
+  const { productId } = await params;
+
+  // Validate that product exists
+  const { data: product, error: productError } = await supabase
+    .from('workspace_products')
+    .select('id')
+    .eq('id', productId)
+    .single();
+
+  if (productError || !product) {
+    return NextResponse.json({ message: 'Product not found' }, { status: 404 });
+  }
+
+  // Get existing inventory to compare
+  const { data: existingInventory, error: fetchError } = await supabase
+    .from('inventory_products')
+    .select('*')
+    .eq('product_id', productId);
+
+  if (fetchError) {
+    console.log(fetchError);
+    return NextResponse.json(
+      { message: 'Error fetching existing inventory' },
+      { status: 500 }
+    );
+  }
+
+  // If inventory is empty, this means unlimited stock - delete all existing inventory
+  if (inventory.length === 0) {
+    const { error: deleteError } = await supabase
+      .from('inventory_products')
+      .delete()
+      .eq('product_id', productId);
+
+    if (deleteError) {
+      console.log(deleteError);
+      return NextResponse.json(
+        { message: 'Error setting unlimited stock' },
+        { status: 500 }
+      );
+    }
```

Comment:
Not sure why we are clearing inventory here. The stock change history should still be tracked even if the stock is unlimited right?

Reply:
Will merge this PR for now, will review the behavior tmr.